device/core/src/pnp/properties.ts

```diff
@@ -0,0 +1,81 @@
+import { JSONValue, JSONObject } from '.';
+import { Twin } from '../twin';
+
+export function generateWritablePropertyResponse(value: JSONValue, ackCode: number, ackDescription: string, ackVersion: number): {
+    value: JSONValue,
+    ac: number,
+    ad: string,
+    av: number
+} {
+    return {
+        value,
+        ac: ackCode,
+        ad: ackDescription,
+        av: ackVersion
+    };
+}
+
+export class ClientPropertyCollection {
+    /**
+     * The object representation of the properties.
+     */
+    backingObject: JSONObject = {};
+
+    constructor(backingObject?: JSONObject) {
+        this.backingObject = backingObject;
```

Comment:
>backingObject [](http://example.com/codeflow?start=29&length=13)

I assume this syntax leaves `this.backingObject == undefined` if you don't pass in an object.  Would `this.backingObject = backingObject or {}` be better?

Reply:
Yes, I just noticed that as well and it's fixed now. Good catch.